lib/landrush/server.rb

```diff
@@ -83,9 +155,90 @@ def self.run
 
         # Default DNS handler
         otherwise do |transaction|
+          # @logger.info "Passing on to upstream: #{transaction.to_s}"
           transaction.passthrough!(server.upstream)
         end
       end
     end
+
+    def self.check_a_record(host, transaction)
+      value = Store.hosts.get(host)
+      if value.nil?
+        return
+      end
+
+      if (IPAddr.new(value) rescue nil)
+        name = transaction.name =~ /#{host}/ ? transaction.name : host
+        transaction.respond!(value, {:ttl => 0, :name => name})
+      else
+        transaction.respond!(Name.create(value), resource_class: IN::CNAME, ttl: 0)
+        check_a_record(value, transaction)
+      end
+    end
+
+    # private methods
+    def self.write_pid(pid)
+      ensure_path_exits(pid_file)
+      File.open(pid_file, 'w') {|f| f << pid.to_s}
```

Comment:
Another smallish style thing: `{ |f| f << pid.to_s }` instead of `{|f| f << pid.to_s}`